test/integration/command/config_test.py

```diff
@@ -232,7 +232,8 @@ def test_config_clean(storage_path):
     assert "bar" not in tc.out
     tc.run("config show core.upload:retry")
     assert "7" not in tc.out
-    assert not os.path.exists(os.path.join(tc.cache_folder, "extensions"))
+    assert os.path.exists(os.path.join(tc.cache_folder, "extensions"))
+    assert not os.path.exists(os.path.join(tc.cache_folder, "extensions", "compatibility", "mycomp.py"))
```

Comment:
This shows that we regenerated the folder (which gets removed), but that the extension file is not there anymore